lib/minify.js

```diff
@@ -271,18 +271,15 @@ async function minify(files, options, _fs_module) {
     }
     if (!HOP(options.format, "code") || options.format.code) {
         if (options.sourceMap) {
-            options.format.source_map = await SourceMap({
+            if (options.sourceMap.includeSources && files instanceof AST_Toplevel) {
+                throw new Error("original source content unavailable");
+            }
+            options.format.source_map = SourceMap({
                 file: options.sourceMap.filename,
                 orig: options.sourceMap.content,
-                root: options.sourceMap.root
+                root: options.sourceMap.root,
+                files: options.sourceMap.includeSources ? files : null,
             });
-            if (options.sourceMap.includeSources) {
```

Comment:
This section was moved into the `SourceMap`. Before, calling `setSourceContent` for the original source contents and the input contents would mean there are 2 contents stored because they have different file names. `SourceMapGenerator` would traverse all mappings and determine if a particular file isn't referenced in the mappings, and would exclude that file's contents from the output.

`GenMapping` doesn't do that traversal. If you tell it that `file.js` has contents, then `file.js` will be in the output map even if it's not actually referenced by any mappings. By moving this into `SourceMap`, we can do a property lookup for source contents as they're needed.